### PR TITLE
fix bug: invalid customized metrics

### PR DESCRIPTION
**Pull request by pfliu-nlp**
(This is a relevant issue that's worthwhile to read: https://github.com/neulab/ExplainaBoard/issues/455)
## Debugging route:
* specified metrics in ExplainaBoard web cannot work 
* verify the correctness of frontend code
* move to SDK
* the same thing happened in SDK CLI
* bug in processor
    * metric_configs should be a list here: https://github.com/neulab/ExplainaBoard/blob/cd54c1b61e490295db8c1cfee8460aff4cce1880/explainaboard/processors/processor.py#L454
    * this will fail: https://github.com/neulab/ExplainaBoard/blob/cd54c1b61e490295db8c1cfee8460aff4cce1880/explainaboard/processors/processor.py#L203

Relevant issues: https://github.com/neulab/explainaboard_web/issues/284

### CODE REVIEW

**Comment by neubig** [comment]
@pfliu-nlp I haven't dug into this in detail yet, but it's not clear to me if the discussion in https://github.com/neulab/ExplainaBoard/issues/455 has finished? Once you've finished and decided which direction to go I'll be happy to review.

**Comment by pfliu-nlp** [comment]
I want you to have a quick read of [this], I think we need to use `Accuracy` instead of `AccuracyConfig` to keep consistent with metric names in the processor. However, to this end, we need to modify this [line](https://github.com/neulab/ExplainaBoard/blob/cd54c1b61e490295db8c1cfee8460aff4cce1880/explainaboard/metrics/registry.py#L13) of code.  Not sure if it makes sense to you.

**Review by neubig — COMMENTED** [review]
Just a note that I'll review this once the discussion in #455 is resolved. Please re-request review then!

**Comment by pfliu-nlp** [comment]
Revised the PR a little bit. 
* register metrics with the form of  `XConfig`
* if users specify metrics, then the default metrics list will be ignored.

File: explainaboard/processors/processor.py [hunk 1]
```diff
@@ -451,7 +456,9 @@ def get_overall_statistics(
         # declare customized features: _features will be updated
         custom_features: dict = metadata.get('custom_features', {})
         custom_analyses: list = metadata.get('custom_analyses', [])
-        metric_configs: dict = metadata.get('metric_configs', {})
+
+        metric_configs: dict = {"example": metadata.get('metric_configs', [])}
```
Comment (odashi) [inline explainaboard/processors/processor.py:460]: Annotate full type information. Partial typing is not useful to detect errors.
```suggestion
        metric_configs: dict[str, list[MetricConfig]] = {"example": metadata.get('metric_configs', [])}
```

File: explainaboard/processors/processor.py [hunk 2]
```diff
@@ -200,12 +200,17 @@ def _customize_analyses(
...
-            if metric_configs and level.name in metric_configs:
-                for metric_config in metric_configs[level.name]:
-                    level.metric_configs.append(metric_config)
+            configs = cast(Dict[str, List], metric_configs)
+            if level.name in configs:
+                for ind, metric_config in enumerate(configs[level.name]):
+                    if ind == 0:
+                        level.metric_configs = [metric_config]
+                    else:
+                        level.metric_configs.append(metric_config)
```
[the remark this answers is not in the document]
Comment (pfliu-nlp) [reply explainaboard/processors/processor.py:209]: thanks! it makes sense

**Review by odashi — APPROVED** [review]
great, thanks!

**Pull request merged**